File: C# Examples/04 Controlling Options/background/SolidWorksMacro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace background
{
    public partial class SolidWorksMacro
    {
public void Main()
{
    Boolean result;
    ModelDoc2 swModel;
    swModel = ((ModelDoc2)(swApp.ActiveDoc));
    int bgd = swApp.GetUserPreferenceIntegerValue((int)
        swUserPreferenceIntegerValue_e.swColorsBackgroundAppearance);

    if(bgd == (int)swColorsBackgroundAppearance_e
                .swColorsBackgroundAppearance_DocumentScene)
    {
        result = swApp.SetUserPreferenceIntegerValue((int)
            swUserPreferenceIntegerValue_e.swColorsBackgroundAppearance, 
            (int)swColorsBackgroundAppearance_e
            .swColorsBackgroundAppearance_DocumentScene);
    }
    swModel.GraphicsRedraw2();
    return;
}

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 04 Controlling Options/background/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System


Partial Class SolidWorksMacro
  Public Sub main()
    Dim result As Boolean
    Dim swModel As ModelDoc2 = Nothing
    swModel = swApp.ActiveDoc

    If swApp.GetUserPreferenceIntegerValue(
  swUserPreferenceIntegerValue_e.swColorsBackgroundAppearance) =
  swColorsBackgroundAppearance_e.swColorsBackgroundAppearance_DocumentScene _
  Then

      result = swApp.SetUserPreferenceIntegerValue(
    swUserPreferenceIntegerValue_e.swColorsBackgroundAppearance,
    swColorsBackgroundAppearance_e.swColorsBackgroundAppearance_Plain)

    Else
      result = swApp.SetUserPreferenceIntegerValue _
      (swUserPreferenceIntegerValue_e.swColorsBackgroundAppearance,
      swColorsBackgroundAppearance_e.
      swColorsBackgroundAppearance_DocumentScene)

    End If

    swModel.GrahicsRedraw2
  End Sub
  ''' <summary>
  ''' The SldWorks swApp variable is pre-assigned for you.
  ''' </summary>
  Public swApp As SldWorks
End Class
















